src/harness/util.py

```diff
@@ -279,3 +279,129 @@ def convertRequestToRequestWithCpiSignature(private_key, cpi_id,
   request['cpiSignatureData']['protectedHeader'] = jwt_message[0]
   request['cpiSignatureData']['encodedCpiSignedData'] = jwt_message[1]
   request['cpiSignatureData']['digitalSignature'] = jwt_message[2]
+
+def getPPinPPA(boundarysPoints, ARCSEC):
+    import numpy as np
+    from shapely import geometry
+    
+    poly = geometry.Polygon([[lon,lat] for lon,lat in boundarysPoints])
+    arcsec = ARCSEC/3600.0
+    maxlat = -90 
+    minlat = 90
+    maxlon = -180
+    minlon = 180
+       
+    for point in boundarysPoints:
+        if point[1] > maxlat:
+            maxlat = point[1]
+        if point[1] < minlat:
+            minlat = point[1]         
+        if point[0] > maxlon:
+            maxlon = point[0]
+        if point[0] < minlon:
+            minlon = point[0]
+    
+    iLat = np.floor(minlat)
+    iLon = np.floor(minlon)
+    nL = np.floor((minlat - iLat)/arcsec)
+    nH = np.ceil((maxlat - iLat)/arcsec)
+    mL = np.floor((minlon - iLon)/arcsec)
+    mH = np.ceil((maxlon - iLon)/arcsec)
+        
+    latgrid = np.arange(iLat + nL*arcsec, iLat + nH *arcsec, arcsec)
+    longrid = np.arange(iLon + mL * arcsec, iLon + mH *arcsec, arcsec)
+    
+    if len(latgrid) > 0 and len(latgrid)>0:
+        ppa = []
+
+        for lat in latgrid:
+            for lon in longrid:
+                ppa_check_point = Point(lon, lat)
+                if ppa_check_point.within(poly):
+                        ppa.extend([[lon, lat]])
+
+        if (len(ppa) == 1):
+            return 0, ppa[0][0], ppa[0][1]
+        else:
+            return -1, -1, -1
+    else:
+        return -1, -1, -1
+
+def QueryPropagationAntennaModel(requestJson):
+    import os
+    from reference_models.propagation import wf_itm
+    from reference_models.propagation import wf_hybrid
+    from reference_models.antenna import antenna
+    from reference_models.geo import vincenty
+    from reference_models.geo import nlcd
+    TERRAIN_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'ned')
+    NLCD_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'nlcd')
+    ITU_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'itu')
+    wf_itm.ConfigureTerrainDriver(terrain_dir=TERRAIN_TEST_DIR)
+    wf_itm.ConfigureItuDrivers(itu_dir=ITU_TEST_DIR)
+    wf_hybrid.ConfigureTerrainDriver(terrain_dir=TERRAIN_TEST_DIR)
+    wf_hybrid.wf_itm.ConfigureItuDrivers(itu_dir=ITU_TEST_DIR)
+    nlcd_driver = nlcd.NlcdDriver(NLCD_TEST_DIR)
```

Comment:
Why all the previous initialization code within the function:
 - all imports should be on top of the file
 - Terrain driver configuration should be done once at the initialization of the testing framework.
 - NLCD driver should be initialized also in the testing framework globally, either as a global or as an attribute in the top class. Currently you are building a NLCD driver for each call, meaning that all call to the NLCD will need to reload the tile, and cannot reuse tile in memory.